salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +53,102 @@ def node_info(node, ca_minion):
     return result
 
 
+def get_cluster_version(api_client):
+    coreV1 = kubernetes.client.CoreV1Api(api_client=api_client)
+    try:
+        namespace = coreV1.read_namespace(name="kube-system")
+    except (ApiException, HTTPError) as exc:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to read namespace information {}'.format(exc)
+        )
+    annotations = namespace.metadata.annotations
+    annotation_key = 'metalk8s.scality.com/cluster-version'
+    if not annotations or annotation_key not in annotations:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return namespace.metadata.annotations[annotation_key]
+
+
+def iso_timestamp_converter(timestamp):
+    if not timestamp:
+        return None
+    return timestamp.isoformat()
+
+
+def get_storage_classes(api_client):
+    storageV1 = kubernetes.client.StorageV1Api(api_client=api_client)
+    try:
+        storageclass_list = storageV1.list_storage_class()
+    except (ApiException, HTTPError) as exc:
+        error_tplt = 'Exception while calling StorageV1->list_storage_class {}'
+        return __utils__['pillar_utils.errors_to_dict'](
+            [error_tplt.format(exc)]
+        )
+
+    storage_classes = {}
+    for storageclass in storageclass_list.items:
+        # Need to convert the datetime object in storageclass to ISO format in
+        # order to make them serializable.
+        storageclass.metadata.creation_timestamp = iso_timestamp_converter(
+            storageclass.metadata.creation_timestamp
+        )
+        storageclass.metadata.deletion_timestamp = iso_timestamp_converter(
+            storageclass.metadata.deletion_timestamp
+        )
+        storage_classes[storageclass.metadata.name] = storageclass.to_dict()
+
+    return storage_classes
+
+
+def list_volumes(api_client, minion_id):
+    customObjectsApi = kubernetes.client.CustomObjectsApi(
+        api_client=api_client
+    )
+    storage_classes = get_storage_classes(api_client)
```

Comment:
With your current implementation, this may be a dict with `_errors` inside. So, not going to work well in case of failure.

Here's what needs to be done:

1. Remove the `try/except` logic from `get_storage_classes`
2. Add a `try/except` in this function:

```
def list_volumes(api_client, minion_id):
    # ...
    try:
        storage_classes = get_storage_classes(api_client)
    except (APIException, HTTPError) as exc:
        return __utils__['...'](error_tplt.format(exc))

    # keep on with the current body
```